modules/mesh-layers/src/utils/matrix.js

```diff
@@ -0,0 +1,125 @@
+import {Matrix4} from 'math.gl';
+
+const RADIAN_PER_DEGREE = Math.PI / 180;
+
+export const matrixLayerProps = {
+  // yaw, pitch and roll are in degrees
+  // https://en.wikipedia.org/wiki/Euler_angles
+  getYaw: {type: 'accessor', value: x => x.yaw || x.angle || 0},
+  getPitch: {type: 'accessor', value: x => x.pitch || 0},
+  getRoll: {type: 'accessor', value: x => x.roll || 0},
+  getScale: {type: 'accessor', value: x => x.scale || [1, 1, 1]},
+  getTranslation: {type: 'accessor', value: x => x.translate || [0, 0, 0]},
+  getMatrix: {type: 'accessor', value: x => x.matrix || null}
+};
+
+const rotationMatrix = new Float32Array(16);
+const modelMatrix = new Matrix4();
+
+function updateRotationMatrix({roll, pitch, yaw}) {
+  const sr = Math.sin(roll);
+  const sp = Math.sin(pitch);
+  const sw = Math.sin(yaw);
+
+  const cr = Math.cos(roll);
+  const cp = Math.cos(pitch);
+  const cw = Math.cos(yaw);
+
+  rotationMatrix[0] = cw * cp; // 0,0
+  rotationMatrix[1] = sw * cp; // 1,0
+  rotationMatrix[2] = -sp; // 2,0
+  rotationMatrix[3] = 0;
+  rotationMatrix[4] = -sw * cr + cw * sp * sr; // 0,1
+  rotationMatrix[5] = cw * cr + sw * sp * sr; // 1,1
+  rotationMatrix[6] = cp * sr; // 2,1
+  rotationMatrix[7] = 0;
+  rotationMatrix[8] = sw * sr + cw * sp * cr; // 0,2
+  rotationMatrix[9] = -cw * sr + sw * sp * cr; // 1,2
+  rotationMatrix[10] = cp * cr; // 2,2
+  rotationMatrix[11] = 0;
+  rotationMatrix[12] = 0;
+  rotationMatrix[13] = 0;
+  rotationMatrix[14] = 0;
+  rotationMatrix[15] = 1;
+}
+
+function makeMatrix({object, getYaw, getPitch, getRoll, getScale, getTranslation}) {
```

Comment:
Tip: You can accept a `targetMatrix` here, then you can avoid a global variable (you move it to into the caller).